frontend/src/components/LogViewer.tsx

```
@@ -0,0 +1,309 @@
+export default LogViewer
+
+export function handleEvent(
+  event: string,
+  lastCircuitId: number,
+  dispatch: any,
+) {
+  // parse the event data EVENT:{event_type, data}:ENDEVENT
+  const eventData = event.match(/EVENT:(.*):ENDEVENT/)
+  if (eventData && eventData[1]) {
+    try {
+      const parsedData = JSON.parse(eventData[1])
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)